### PR TITLE
fix: add recommended resource allocation

### PR DESCRIPTION
**Pull request by qixiang-liu**
#### What type of PR is this?
fix

#### What this PR does / why we need it:

The resource configuration can be adjusted

#### Which issue(s) this PR fixes:

Fixes #
https://github.com/gocrane/crane/issues/698

### CODE REVIEW

File: deploy/craned/deployment.yaml [hunk 1]
```diff
@@ -115,6 +115,13 @@ data:
             apiVersion: apps/v1
           - kind: StatefulSet
             apiVersion: apps/v1
+        config:
```
Comment (qmhu) [inline deploy/craned/deployment.yaml:118]: no needed

File: examples/analytics/recommendation-configuration.yaml [hunk 2]
```diff
@@ -13,6 +13,13 @@ recommenders:
         apiVersion: apps/v1
       - kind: StatefulSet
         apiVersion: apps/v1
+    config:
```
Comment (qmhu) [inline examples/analytics/recommendation-configuration.yaml:16]: also no needed

File: pkg/recommendation/recommender/resource/recommend.go [hunk 3]
```diff
@@ -57,9 +57,9 @@ func (rr *ResourceRecommender) makeCpuConfig() *config.Config {
 			TargetUtilization: rr.CpuTargetUtilization,
 			Percentile:        rr.CpuRequestPercentile,
 			Histogram: predictionapi.HistogramConfig{
-				HalfLife:   "24h",
-				BucketSize: "0.1",
-				MaxValue:   "100",
+				HalfLife:   rr.Recommender.Config["cpu-histogram-half-life"],
```
Comment (qmhu) [inline pkg/recommendation/recommender/resource/recommend.go:60]: please refer to other configuration's implement in register.go

File: pkg/recommendation/recommender/resource/recommend.go [hunk 4]
```diff
@@ -75,9 +75,9 @@ func (rr *ResourceRecommender) makeMemConfig() *config.Config {
 			Percentile:        rr.MemPercentile,
 			TargetUtilization: rr.MemTargetUtilization,
 			Histogram: predictionapi.HistogramConfig{
-				HalfLife:   "48h",
-				BucketSize: "104857600",
-				MaxValue:   "104857600000",
+				HalfLife:   rr.Recommender.Config["mem-histogram-half-life"],
```
Comment (qmhu) [inline pkg/recommendation/recommender/resource/recommend.go:78]: the same above

File: site/content/en/docs/Tutorials/Recommendation/recommendation-framework.md [hunk 5]
```diff
@@ -166,10 +166,17 @@ data:
             apiVersion: apps/v1
           - kind: StatefulSet
             apiVersion: apps/v1
+        config:                              # settings for recommender
```
Comment (qmhu) [inline site/content/en/docs/Tutorials/Recommendation/recommendation-framework.md:169]: No need to add here, better to update here :https://github.com/gocrane/crane/blob/main/site/content/en/docs/Tutorials/Recommendation/resource-recommendation.md
configuration part

File: site/content/zh/docs/Tutorials/Recommendation/recommendation-framework.md [hunk 6]
```diff
@@ -166,6 +166,13 @@ data:
             apiVersion: apps/v1
           - kind: StatefulSet
             apiVersion: apps/v1
+        config:                              # settings for recommender
```
Comment (qmhu) [inline site/content/zh/docs/Tutorials/Recommendation/recommendation-framework.md:169]: the same above

**Comment by qixiang-liu** [comment]
All adjustments have been made according to the specification

File: pkg/recommendation/recommender/resource/recommend.go [hunk 7]
```diff
@@ -48,6 +48,18 @@ func (rr *ResourceRecommender) PreRecommend(ctx *framework.RecommendationContext
 }
 
 func (rr *ResourceRecommender) makeCpuConfig() *config.Config {
+	cpuHistogramHalfLife, exists := rr.Recommender.Config["cpu-histogram-half-life"]
```
Comment (qmhu) [inline pkg/recommendation/recommender/resource/recommend.go:51]: Please move it to register.go , we use there to centralized management the configurations.

File: site/content/zh/docs/Tutorials/Recommendation/resource-recommendation.md [hunk 8]
```diff
@@ -149,22 +149,28 @@ container_memory_working_set_bytes{container!="POD",namespace="crane-system",pod
...
+| mem-request-margin-fraction | 0.15         | Memory 推荐值扩大系数，0.15指推荐值乘以 1.15   |
+| mem-target-utilization      | 1            | Memory 目标利用率，0.8 指推荐值除以 0.8      |
+| specification               | false        | 是否开启资源规格规整                       |
+| specification-config        | ""           | 资源规格，注意格式，详细的默认配置请见下方表格          |
+| oom-protection              | true         | 是否开启 OOM 保护                      |
+| oom-history-length          | 168h         | OOM 历史事件的事件，过期事件会被忽略             |
+| oom-bump-ratio              | 1.2          | OOM 内存放大系数                       |
+| cpu-histogram-half-life     | 24h          | cpu检查的周期                         |
+| cpu-histogram-bucket-size   | 0.1          | cpu最小请求的值（resources.request.cpu） |
+| cpu-histogram-max-value     | 100          | cpu最大请求的值（resources.limits.cpu）  |
```
Comment (qmhu) [inline site/content/zh/docs/Tutorials/Recommendation/resource-recommendation.md:170]: 这里字段的含义需要修改
cpu-histogram-bucket-size，均衡桶的大小，同时也等于cpu推荐的最小值
cpu-histogram-max-value ，均衡桶的最大值，同时也等于cpu推荐的最大值
half-life  可以去掉，目前没用到，包括前面的配置代码也去掉吧

**Comment by qixiang-liu** [comment]
All adjustments have been made according to the specification

**Review by qmhu — APPROVED** [review]

**Pull request merged**